Material.Blazor/Components/MenuSurface/MBMenuSurface.razor.cs

```diff
@@ -0,0 +1,112 @@
+
+
+        /// <summary>
+        /// Regular, fullwidth or fixed positioning/width.
+        /// </summary>
+        [Parameter] public MBMenuSurfacePositioning MenuSurfacePositioning { get; set; } = MBMenuSurfacePositioning.Regular;
+
+
+        private DotNetObjectReference<MBMenuSurface> ObjectReference { get; set; }
+        private ElementReference ElementReference { get; set; }
+        private bool IsOpen { get; set; } = false;
+
+
+        // Would like to use <inheritdoc/> however DocFX cannot resolve to references outside Material.Blazor
+        protected override async Task OnInitializedAsync()
+        {
+            await base.OnInitializedAsync();
+
+            ConditionalCssClasses
+                .AddIf(GetMenuSurfacePositioningClass(MenuSurfacePositioning), () => MenuSurfacePositioning != MBMenuSurfacePositioning.Regular);
+
+            ObjectReference = DotNetObjectReference.Create(this);
+        }
+
+
+        private bool _disposed = false;
+        protected override void Dispose(bool disposing)
+        {
+            if (_disposed)
+            {
+                return;
+            }
+
+            if (disposing)
+            {
+                ObjectReference?.Dispose();
+            }
+
+            _disposed = true;
+
+            base.Dispose(disposing);
+        }
+
+
+        /// <summary>
+        /// For Material Theme to notify of menu closure via JS Interop.
+        /// </summary>
+        [JSInvokable("NotifyClosedAsync")]
+        //[System.Diagnostics.CodeAnalysis.SuppressMessage("Style", "IDE0060:Remove unused parameter", Justification = "<Pending>")]
+        public async Task NotifyClosedAsync()
+        {
+            IsOpen = false;
+            await Task.CompletedTask;
+        }
+
+
+        /// <summary>
+        /// Toggles the menu open and closed. NEED TO RETURN <code>Task</code> RATHER THAN <code>Task&lt;string&gt;</code> IN VERSION 2.0.0
+        /// </summary>
+        /// <returns></returns>
+        public async Task ToggleAsync()
+        {
+            if (IsOpen)
+            {
+                await JsRuntime.InvokeVoidAsync("MaterialBlazor.MBMenuSurface.hide", ElementReference);
+                IsOpen = false;
+            }
+            else
+            {
+                await JsRuntime.InvokeVoidAsync("MaterialBlazor.MBMenuSurface.show", ElementReference);
+                IsOpen = true;
+            }
+        }
+
+
+        /// <inheritdoc/>
+        private protected override async Task InstantiateMcwComponent() => await JsRuntime.InvokeVoidAsync("MaterialBlazor.MBMenuSurface.init", ElementReference, ObjectReference);
+
+
+        /// <summary>
+        /// Returns a menu surface class determined by the parameter.
+        /// </summary>
+        /// <param name="surfacePositioning"></param>
+        /// <returns></returns>
+        internal static string GetMenuSurfacePositioningClass(MBMenuSurfacePositioning surfacePositioning) =>
```

Comment:
Consider declaring this private. I can't see why this is necessary outside this component.